File: medplum-typescript/packages/sdk/src/model/fhir/MeasureReport.ts
import { canonical } from "./canonical";
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { MeasurereportStatus } from "./MeasurereportStatus";
import { MeasurereportType } from "./MeasurereportType";
import { MeasureReport_Group } from "./MeasureReport_Group";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface MeasureReport {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    status: MeasurereportStatus | null | undefined;
    type: MeasurereportType | null | undefined;
    measure: canonical;
    subject: Reference | null | undefined;
    date: dateTime | null | undefined;
    reporter: Reference | null | undefined;
    period: Period;
    improvementNotation: CodeableConcept | null | undefined;
    group: MeasureReport_Group[] | null | undefined;
    evaluatedResource: Reference[] | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/MeasureReport.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMeasureReport.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MeasureReport {
  Optional<dateTime> date();

  Optional<List<MeasureReport_Group>> group();

  Optional<List<Reference>> evaluatedResource();

  Optional<MeasurereportType> type();

  Optional<Narrative> text();

  Optional<List<ResourceList>> contained();

  Optional<MeasurereportStatus> status();

  Optional<id> id();

  Optional<Reference> reporter();

  canonical measure();

  Optional<code> language();

  Optional<CodeableConcept> improvementNotation();

  Optional<List<Extension>> extension();

  Optional<List<Identifier>> identifier();

  Optional<uri> implicitRules();

  Period period();

  Optional<Reference> subject();

  String resourceType();

  Optional<Meta> meta();

  Optional<List<Extension>> modifierExtension();

  static ImmutableMeasureReport.MeasureBuildStage builder() {
    return ImmutableMeasureReport.builder();
  }
}
